src/ucp/core/ucp_ep.c

```diff
@@ -523,24 +527,32 @@ ucp_ep_create_api_conn_request(ucp_worker_h worker,
     }
 
     if (status == UCS_OK) {
-        *ep_p = ep;
         goto out;
     }
 
 out_ep_destroy:
     ucp_ep_destroy_internal(ep);
 out:
-    if (!ucp_worker_sockaddr_is_cm_proto(worker)) {
-        if (status == UCS_OK) {
+    if (status == UCS_OK) {
+        if (!ucp_worker_sockaddr_is_cm_proto(worker)) {
```

Comment:
when do we send accept for cm proto?

Reply:
uct_listener doesn't have similar API, instead, we create uct_ep in `ucp_ep_cm_connect_server_lane`
 https://github.com/evgeny-leksikov/ucx/blob/3d4b43459c64015c930084754c5cf146545cfe36/src/ucp/wireup/wireup_cm.c#L535